src/core/server/ui_settings/ui_settings_client.ts

```diff
@@ -270,4 +272,19 @@ export class UiSettingsClient implements IUiSettingsClient {
       (ignore401Errors && isNotAuthorizedError(error))
     );
   }
+
+  // TODO: [RENAMEME] Temporary code for backwards compatibility.
+  // https://github.com/opensearch-project/OpenSearch-Dashboards/issues/334
+  private translateChanges(changes: Record<string, any>, source: string, dest: string) {
+    const translatedChanges: Record<string, any> = {};
+    Object.keys(changes).forEach((key) => {
+      let translatedKey = key;
+      const value = changes[key];
+      if (key.includes(source)) {
+        translatedKey = key.replace(source, dest);
+      }
+      translatedChanges[translatedKey] = value;
+    });
+    return translatedChanges;
```

Comment:
I know this is temporary code, but can't we simplify?

```typescript
private translateChanges(changes: Record<string, any>, source: string, dest: string) {
  return Objects.keys(changes)
    .filter(key => key.includes(source))
    .reduce((translatedChanges, key) => {
      translatedChanges[key.replace(source, dest)] = changes[key];
      return translatedChanges;
    }, {});
}
```

Reply:
This would filter any changes that don't include the source right? Probably can just be 

```
private translateChanges(changes: Record<string, any>, source: string, dest: string) {
  return Objects.keys(changes)
    .reduce((translatedChanges, key) => {
      translatedChanges[key.replace(source, dest)] = changes[key];
      return translatedChanges;
    }, {});
}
```

didn't put much thought into this code as long as is it worked but better not to sacrifice simplicity .

can update it.

Reply:
Oh yes, you're correct. I misunderstood the original logic. We don't want to filter out all of the other keys. Yeah looks good.